src/client/Microsoft.Identity.Client/TelemetryCore/OpenTelemetry/OpenTelemetryClient.cs

```diff
@@ -70,25 +76,57 @@ public class OpenTelemetryClient
             unit: "ms",
             description: "Performance of token acquisition calls network latency");
 
-        internal static readonly ActivitySource s_acquireTokenActivity = new ActivitySource(ActivitySourceName, "1.0.0");
+        internal static readonly Lazy<Activity> s_activity = new Lazy<Activity>(() => s_acquireTokenActivity.StartActivity("Token Acquisition", ActivityKind.Internal));
+#endif
+
+        void IOpenTelemetryClient.LogActivity(Dictionary<string, object> tags)
+        {
+#if NETSTANDARD || NET6_0 || NET462
+            foreach (KeyValuePair<string, object> tag in tags)
+            {
+                s_activity.Value?.AddTag(tag.Key, tag.Value);
+            }
+#endif
+        }
+
+        void IOpenTelemetryClient.LogActivityStatus(bool success)
+        {
+#if NETSTANDARD || NET6_0 || NET462
```

Comment:
There is a more elegant way of doing this - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/build/platform_and_feature_flags.props

(also, enable it on NET6-Windows)